server/swap/swap.go

```diff
@@ -1122,33 +1154,35 @@ func (s *Swapper) checkInactionBlockBased(assetID uint32) {
 		log.Tracef("checkInactionBlockBased: asset %d, match %v (%v)",
 			assetID, match.ID(), match.Status)
 
-		failMatch := func() {
+		deleteMatch := func() {
 			// Fail the match, and assign fault if lock times are not passed.
-			s.failMatch(match, !match.expiredBy(now))
-			deletions = append(deletions, match)
+			s.deleteMatch(match)
+			failures = append(failures, fail{match, !match.expiredBy(now)})
```

Comment:
Yeah, and `failMatch` only looks at the Status field.  Any updates to Status are intentionally done with the matches map locked in processInit and processRedeem.  It wouldn't hurt for failMatch to use the matchTracker mutex too but it wasn't needed and still isn't I'm fairly certain.

`processBlock` -> `tryConfirmSwap` could still have a reference to the match, but it's not touching the `matchTracker.Status` field that `matchTracker.mtx` is assigned to.  In retrospect, I think we could have put the mutex in `order.Match` instead of `swap.matchTracker`.